src/modules/twap/hooks/useTwapOrdersExecutions.ts

```diff
@@ -68,11 +71,63 @@ export function useTwapOrdersExecutions(ids: string[]): TwapOrdersExecutionMap {
 
       return acc
     }, {})
-  }, [ids, partSets, allPartOrders])
+  }, [allOrdersInfo, partSets, allDiscreteOrders])
 }
 
 function sumChildrenAmount(children: Order[], key: keyof OrderInfoApi): BigInt {
   return children.reduce((acc, order) => {
     return acc + BigInt((order.apiAdditionalInfo?.[key] || '0') as string)
   }, BigInt(0))
 }
+
+/**
+ * There might be a case when a TWAP order consists of 3 orders (for example)
+ * First two parts are expired (never been executed or event created in order-book)
+ * And the last part is filled
+ * The example on the picture: https://user-images.githubusercontent.com/7122625/258057476-19022558-5eb8-4e93-8afc-afd186bfbc30.png
+ *
+ * In this case the discreteOrders array will contain only one order (the last one)
+ * So, we need to calculate the number of confirmed parts based on the time passed
+ */
+function getConfirmedPartsCount(twapOrderInfo: TwapOrderInfo, discreteOrders: Order[]): number {
```

Comment:
This is all a bit complex, but I can understand it.
The comments are very helpful

Reply:
The case is a bit complex :)